tests/unit/s2n_tls_hybrid_prf_test.c

```diff
@@ -135,22 +110,35 @@ int main(int argc, char **argv)
         s2n_stuffer_write(&combined_stuffer, &classic_pms);
         s2n_stuffer_write(&combined_stuffer, &kem_pms);
 
-        EXPECT_SUCCESS(convert_hex_to_bytes(conn->secure.client_random, client_random_hex_in));
-        EXPECT_SUCCESS(convert_hex_to_bytes(conn->secure.server_random, server_random_hex_in));
-
-        EXPECT_SUCCESS(s2n_alloc(&conn->secure.client_key_exchange_message, (strlen(client_key_exchange_message_1) + strlen(client_key_exchange_message_2)) / 2));
-        const char *strings[] = {client_key_exchange_message_1, client_key_exchange_message_2};
-        EXPECT_SUCCESS(multi_convert_hex_to_bytes(conn->secure.client_key_exchange_message.data, strings, 2));
-
+        /* Prove to ourselves that client_random and server_random are the same length
+         * so that we can combine both assignments in a single loop. */
+        eq_check(CLIENT_RANDOM_LENGTH, SERVER_RANDOM_LENGTH);
+        for (uint32_t j = 0; j < CLIENT_RANDOM_LENGTH; j++) {
+            conn->secure.client_random[j] = client_random[j];
+            conn->secure.server_random[j] = server_random[j];
+        }
 
-        uint8_t expected_master_secret[S2N_TLS_SECRET_LEN];
-        EXPECT_SUCCESS(convert_hex_to_bytes(expected_master_secret, expected_master_secret_hex_in));
+        EXPECT_SUCCESS(s2n_alloc(&conn->secure.client_key_exchange_message, client_key_exchange_message_length));
+        for (uint32_t j = 0; j < client_key_exchange_message_length; j++) {
+            conn->secure.client_key_exchange_message.data[j] = client_key_exchange_message[j];
+        }
 
         EXPECT_SUCCESS(s2n_hybrid_prf_master_secret(conn, &combined_pms));
-
         EXPECT_BYTEARRAY_EQUAL(expected_master_secret, conn->secure.master_secret, S2N_TLS_SECRET_LEN);
         EXPECT_SUCCESS(s2n_free(&conn->secure.client_key_exchange_message));
         EXPECT_SUCCESS(s2n_connection_free(conn));
+
+        free(premaster_kem_secret);
+        free(client_key_exchange_message);
+    }
+
+    free(premaster_classic_secret);
+    free(client_random);
+    free(server_random);
+    free(expected_master_secret);
```

Comment:
NP: can this be cleaned up with defer_cleanup?

Reply:
I realize now that these lengths are fixed and defined so we don't need to `malloc` at all.